python-avd/pyavd/_eos_designs/shared_utils/inband_management.py

```diff
@@ -171,14 +169,13 @@ def inband_management_parent_vlans(self: SharedUtilsProtocol) -> dict:
         svis = {}
         subnets = []
         ipv6_subnets = []
-        peers = natural_sort(get(self.hostvars, f"avd_topology_peers..{self.hostname}", separator="..", default=[]))
-        for peer in peers:
-            peer_facts = self.get_peer_facts(peer, required=True)
-            if (vlan := peer_facts.get("inband_mgmt_vlan")) is None:
+        for peer in self.switch_facts.downlink_switches:
```

Comment:
are we sure losing natural_sort here is not going to have adverse effects? (does not seem to)

Reply:
Worst case vlans are reordered in structured config, but they are sorted again in eos_cli_config_gen. The order of downlink_switches is deterministic, since we are sorting on the fabric switches in the fact phase, so we wont flip structured config back and forth.